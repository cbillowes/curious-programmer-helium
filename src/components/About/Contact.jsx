import React, { Component } from "react"
import ExternalLink from "../Link/ExternalLink"
import socialLinks from "../../../data/SocialLinks"
import "./Contact.scss"

class Contact extends Component {
  render() {
    return (
      <div className="contact">
        {socialLinks.contact.map((item) => {
          return (
            <ExternalLink
              key={`contact-${item.icon}`}
              className={`icon ${item.icon}`}
              to={item.href} />
          )
        })}
      </div>
    )
  }
}

export default Contact
