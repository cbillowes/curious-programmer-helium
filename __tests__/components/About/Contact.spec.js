import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Contact from "../../../src/components/About/Contact"

describe("technologies component", () => {
  it("should render", () => {
    const component = <Contact />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render contents", () => {
    const component = mount(<Contact />)
  })
})