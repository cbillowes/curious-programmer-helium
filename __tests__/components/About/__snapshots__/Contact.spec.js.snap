// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`technologies component should render 1`] = `
<div
  className="contact"
>
  <a
    className="icon email"
    href="/mailto:curiousprogrammer.io@gmail.com"
    onClick={[Function]}
    onMouseEnter={[Function]}
  />
  <a
    className="icon rss"
    href="/rss.xml"
    onClick={[Function]}
    onMouseEnter={[Function]}
  />
  <a
    className="icon twitter"
    href="/https:/twitter.com/cbillowes"
    onClick={[Function]}
    onMouseEnter={[Function]}
  />
  <a
    className="icon stack-overflow"
    href="/https:/stackoverflow.com/users/849986/cbillowes"
    onClick={[Function]}
    onMouseEnter={[Function]}
  />
  <a
    className="icon github"
    href="/https:/github.com/cbillowes"
    onClick={[Function]}
    onMouseEnter={[Function]}
  />
  <a
    className="icon linkedin"
    href="/https:/www.linkedin.com/in/cbouwer/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  />
</div>
`;
